### PR TITLE
DB-9015 Add support for STRING_AGG

### PR DESCRIPTION
**Pull request by dgomezferro**

### CODE REVIEW

**Comment by dgomezferro** [comment]
jenkins please test branch @cdh5.12.2

**Comment by spliceci** [comment]
TEST FAILED

**Comment by dgomezferro** [comment]
jenkins please test branch @cdh5.12.2

**Comment by spliceci** [comment]
TEST FAILED

**Comment by dgomezferro** [comment]
jenkins please test branch @cdh5.12.2

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

File: splice_machine/src/main/java/com/splicemachine/SpliceKryoRegistry.java [hunk 1]
```diff
@@ -960,5 +960,7 @@ public Optional read(Kryo kryo, Input input, Class<Optional> type) {
         instance.register(BroadcastFullOuterJoinOperation.class,EXTERNALIZABLE_SERIALIZER,326);
         instance.register(MergeSortFullOuterJoinOperation.class,EXTERNALIZABLE_SERIALIZER,327);
         instance.register(FakeColumnStatisticsImpl.class,EXTERNALIZABLE_SERIALIZER,328);
+        instance.register(StringAggregator.class,EXTERNALIZABLE_SERIALIZER,329);
+        instance.register(StringBuilder.class,330);
```
Comment (msirek) [inline splice_machine/src/main/java/com/splicemachine/SpliceKryoRegistry.java]: Make sure to also add these to SpliceSparkKryoRegistrator.java.

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/window/function/StringAggregator.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+import com.splicemachine.db.iapi.types.SQLLongint;
+import com.splicemachine.db.iapi.types.SQLVarchar;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+
+/**
+ *
+ * Created by jyuan on 7/31/14.
```
Comment (msirek) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/window/function/StringAggregator.java]: Update the "Created by" text.

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/window/function/StringAggregator.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    }
+
+    @Override
+    public WindowFunction newWindowFunction() {
+        return new StringAggregator(separator);
+    }
+
+    @Override
+    public void writeExternal(ObjectOutput out) throws IOException {
+        out.writeObject(acc);
```
Comment (msirek) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/window/function/StringAggregator.java]: Please consider writing a class version number (e.g. 1) as part of SerDe in case this is every stored on disk.

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/execute/StringAggregator.java [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+    //
+    // EXTERNALIZABLE INTERFACE
+    //
+    /////////////////////////////////////////////////////////////
+
+    /**
+     * @throws IOException on error
+     */
+    public void writeExternal(ObjectOutput out) throws IOException {
+        out.writeObject(aggregator);
```
Comment (msirek) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/execute/StringAggregator.java]: Please consider writing a class version number (e.g. 1) as part of SerDe in case this is every stored on disk.

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/window/function/StringAggregator.java (hunk 2 above)
Comment (jyuanca) [reply splice_machine/src/main/java/com/splicemachine/derby/impl/sql/execute/operations/window/function/StringAggregator.java]: I did not create this file :-)

File: start-splice-cluster [hunk 5]
```diff
@@ -146,6 +146,8 @@ fi
 echo "Running Splice $PROFILE master and ${MEMBERS} regionservers with CHAOS = ${CHAOS} in:"
 echo "   ${RUN_DIR}"
 
+export MAVEN_OPTS="$MAVEN_OPTS -Xmx5000m"
```
Comment (arnaud-splice) [inline start-splice-cluster]: Why is that necessary?

Comment (dgomezferro) [reply start-splice-cluster]: When compiling branch 3.0 maven ran out of memory, I just made the same change across the board.

**Comment by dgomezferro** [comment]
jenkins please test branch @cdh5.12.2

**Comment by spliceci** [comment]
TEST FAILED

**Comment by dgomezferro** [comment]
jenkins please test branch @cdh5.12.2

**Comment by spliceci** [comment]
TEST SUCCEEDED +1
